controllers/components/kserve/kserve_controller_actions.go

```diff
@@ -137,6 +137,21 @@ func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
 	return nil
 }
 
+func renderTemplates(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	k, ok := rr.Instance.(*componentApi.Kserve)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentApi.Kserve)", rr.Instance)
+	}
+
+	data := getTemplateData(ctx, rr.Client, k, rr.DSCI)
+
+	actionFn := template.NewAction(
+		template.WithData(data),
+	)
+
+	return actionFn(ctx, rr)
```

Comment:
this will cause the action to ignore any caching

Reply:
Yeah, this isn't ideal (and it's the reason I didn't bother adding the `template.WithCache()` ActionOpts): ideally we'd be able to add the action directly, rather than having it be recreated on each rr.

I did it this way to be able to have access to the Kserve and DSCI from the rr (and the ctx that gets passed to the action along with the rr at reconcile time), to be able to set the template data.

Do you have any suggestions on how this could be done better? One possibility might be to get the data in an earlier action, and store that somewhere. Is there any reasonable place to store it? The idea of it being some sort of global var makes me a bit nervous.

Reply:
I'll open a PR to add such support in the rendering action

Reply:
@grdryn https://github.com/opendatahub-io/opendatahub-operator/pull/1527